test/plugins/webhooks/github.test.js

```diff
@@ -253,6 +253,16 @@ describe('github plugin test', () => {
                     assert.equal(reply.statusCode, 500);
                 });
             });
+
+            it('handles checkouting when given a non-listed user', () => {
+                userFactoryMock.get.resolves(null);
+                userFactoryMock.get.withArgs({ username: 'sd-buildbot' }).resolves(userMock);
+
+                return server.inject(options)
+                .then((response) => {
+                    assert.equal(response.statusCode, 201);
```

Comment:
I mean with that argument we can remove all the `calledWith` in our tests where `statusCode` is already checked. 
One example is here (and throughout this file): https://github.com/screwdriver-cd/screwdriver/blob/master/test/plugins/builds.test.js#L336-L337

I understand that technically it won't pass the test if that function wasn't called. But since the code says it calls it, we should assert that it's actually called. That way if **somehow** it returns `201` without calling that function, it will fail here. The error message  will be more accurate as well.